🧪 **4. Hypothesis Testing**

In [6]:
# Import libraries
import pandas as pd
import numpy as np

# Create a sample dataset
data = {
    "Test Scores": [67, 85, 90, 76, 88, 92, 70, 78, 82, 94]
}
df = pd.DataFrame(data)



In [7]:
# Import necessary library
from scipy.stats import ttest_1samp, chi2_contingency, f_oneway

# One-sample t-test: Is the mean score different from 80?
t_stat, p_value = ttest_1samp(df["Test Scores"], 80)
print(f"One-sample t-test result: t = {t_stat:.3f}, p = {p_value:.3f}")


One-sample t-test result: t = 0.750, p = 0.472


### 🧪 One-Sample T-Test Analysis

**Test Summary:**
- **Null Hypothesis (H₀):** The population mean is equal to 80.
- **Alternative Hypothesis (H₁):** The population mean is different from 80.
- **t-statistic:** 0.750
- **p-value:** 0.472

---

### 📊 Interpretation:

- The **t-statistic (0.750)** is relatively close to 0, indicating that the sample mean is not far from the hypothesized value of 80.
- The **p-value (0.472)** is **greater than 0.05**, which means the result is **not statistically significant** at the 5% significance level.

---

### ✅ Conclusion:

- **Fail to reject the null hypothesis.**
- There is **no statistically significant difference** between the sample mean and 80.
- This suggests that the average test score in the sample could reasonably be equal to 80.

---

### ℹ️ Note:

If the p-value had been below 0.05, it would indicate strong evidence against the null hypothesis. But in this case, we do not have enough evidence to say the mean differs from 80.


**Chi-Square Example (Contingency Table)**

In [8]:
# Create a contingency table
contingency_data = [[30, 10], [20, 40]]
chi2, p, dof, expected = chi2_contingency(contingency_data)
print(f"Chi-square test result: chi2 = {chi2:.3f}, p = {p:.3f}")


Chi-square test result: chi2 = 15.042, p = 0.000


### 🧮 Chi-Square Test of Independence Analysis

**Test Summary:**
- **Null Hypothesis (H₀):** There is **no association** between the two categorical variables (they are independent).
- **Alternative Hypothesis (H₁):** There **is an association** between the two variables (they are dependent).
- **Chi-square statistic (χ²):** 15.042
- **p-value:** 0.000

---

### 📊 Interpretation:

- The **chi-square value (15.042)** is relatively high, which suggests a large difference between the observed and expected frequencies in the contingency table.
- The **p-value (0.000)** is **well below the 0.05 threshold**, indicating a **highly significant result**.

---

### ✅ Conclusion:

- **Reject the null hypothesis.**
- There is **strong evidence of an association** between the two categorical variables.
- This means the variables are **not independent**—one may influence or be related to the other.

---

### ℹ️ Note:

A very small p-value (like 0.000) strongly suggests that the observed relationship in the contingency table is **unlikely due to chance**. Further investigation can help identify the nature of the association.


**ANOVA Example**

In [9]:
# Simulate test scores from 3 groups
group1 = [67, 70, 72]
group2 = [80, 85, 83]
group3 = [90, 92, 94]

anova_result = f_oneway(group1, group2, group3)
print(f"ANOVA test result: F = {anova_result.statistic:.3f}, p = {anova_result.pvalue:.3f}")


ANOVA test result: F = 67.940, p = 0.000


### 🧪 ANOVA (Analysis of Variance) Test Analysis

**Test Summary:**
- **Null Hypothesis (H₀):** All group means are **equal**.
- **Alternative Hypothesis (H₁):** At least one group mean is **different** from the others.
- **F-statistic:** 67.940
- **p-value:** 0.000

---

### 📊 Interpretation:

- The **F-statistic (67.940)** is quite large, which indicates a high ratio of **between-group variance** to **within-group variance**.
- The **p-value (0.000)** is **much smaller than 0.05**, meaning the result is **highly statistically significant**.

---

### ✅ Conclusion:

- **Reject the null hypothesis.**
- There is strong evidence that **at least one group mean is significantly different** from the others.
- This suggests that the group or treatment effects are **not all the same**, and further post-hoc analysis (like Tukey's test) could identify **which groups differ**.

---

### ℹ️ Note:

A high F-value and low p-value indicate that **variation between groups** is greater than variation within groups, confirming that group membership **influences the outcome variable**.
